# Butyrate producer abundance 
# Dr. Jonathan Golob  has parsed out a list of butyrate producers from article "Impact of gut colonization with butyrate-producing microbiota on respiratory viral infection following allo-HCT" (Blood 2018)

library(ape)
source("mothur_taxonomy_munging_function.R")
source("remove_otu_function.R")

# Read in taxonomy file
tax_table = read.table("NH_microbiota.taxonomy", header = T)
taxonomy = mothur_tax_munging_func(tax_table)


# Read in 16S reads
OTU_counts = read.table("NH_microbiota.opti_mcc.shared", header = TRUE)
rownames(OTU_counts) = OTU_counts$Group
otu_samples = OTU_counts[,grep("^Otu", colnames(OTU_counts))]

# Remove skin-associated OTUs from 16S data
skin_genus = c("Staphylococcus", "Corynebacterium", "Propionibacterium")
skin_depleted_otu_samples = remove_otu(otu_mat = otu_samples, taxonomy = taxonomy, remove_genus = skin_genus)

# Calculate relative abundnace of each OTU
skin_depleted_otu_sample_RA = skin_depleted_otu_samples/rowSums(skin_depleted_otu_samples)

# Load patient data
temp_df = read.csv("patient_data.csv", header = T, row.names = 1)

# Append microbiome data to patient data
pt_mbiome_df = cbind(temp_df, skin_depleted_otu_sample_RA[rownames(temp_df), ])


## Map Jonathan's parsed tax_id file to Blast result which is generated by blasting representative sequence to SILVA database

butyrate = read.csv("butyrate_taxons.csv", header = TRUE)

rep_seq = read.FASTA("NH_microbiota.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.rep.fasta", type = "DNA")

# all_blast = read.table("testblast", sep = "\t") 
# testblast is not available due to its large size;
# code for doing a local blast is available in codes folder called "blast command"

all_blast$spp = sapply(as.character(all_blast$V10), FUN = function(x){
  strsplit(x, ";")[[1]][length(strsplit(x, ";")[[1]])]
})
test_blast = as.matrix(all_blast[, c("V1", "spp")])
test_blast = test_blast[grep("uncultured|unidentified|metagenome|sp.", test_blast[,"spp"], invert = TRUE), ]

unique_blast_contigs = unique(test_blast[,"V1"])

# Linking contig names to bacterial name
blast_hits = sapply(unique_blast_contigs, FUN = function(x){
  
  hit_ind = names(rev(sort(table(sapply(test_blast[test_blast[,"V1"] %in% x,"spp"], FUN = function(x){paste(strsplit(x, " ")[[1]][1:2], collapse = " ")})))))
  hit_ind
  
})

# butyrate blast 
butyrate_spp_blast = sapply(as.character(butyrate$tax_name), FUN = function(b){
  ind = names(which(unlist(lapply(blast_hits, FUN = function(h){sum(h %in% b) > 0}))))
  ind
})

buty_genera = unique(sapply(as.character(butyrate$tax_name), FUN = function(g){
  strsplit(g, " ")[[1]][1]
}))
buty_genera = gsub("[", "", buty_genera, fixed = TRUE)
buty_genera = gsub("]", "", buty_genera, fixed = TRUE)


buty_otus = sapply(names(butyrate_spp_blast), FUN = function(b){
  id = butyrate_spp_blast[[b]]
  otu = sapply(id, FUN = function(i){
    strsplit(strsplit(names(rep_seq)[grep(i, names(rep_seq))], "\t")[[1]][2], "|", fixed = T)[[1]][1]})
  buty_otu = taxonomy[taxonomy$OTU %in% otu & grepl(paste(c("Clostrid", buty_genera), collapse = "|"), taxonomy$genus), "OTU"]
  
})

buty_otus = unique(unlist(buty_otus))

buty_ra = rowSums(skin_depleted_otu_samples[,colnames(skin_depleted_otu_samples) %in% buty_otus])/rowSums(skin_depleted_otu_samples) * 100

pt_mbiome_df$buty_ra = buty_ra[rownames(pt_mbiome_df)]

buty_mod = glm(binary_outcome ~ buty_ra , data = pt_mbiome_df, family = "binomial")


coef(summary(buty_mod)) # results shows for every 1% decrease in butyrate producer relative abundance, there is ~1% decrease risk of acquiring a resistant organism (p = 0.21).
